### PR TITLE
Fixes #6: Added grafana app

### LINKED ISSUE #6: [App request] Grafana
## App

Grafana - for dashboards and monitoring

Name `grafana`

Helm: 3 only

Persistence off by default, but flag available

https://github.com/helm/charts/tree/master/stable/grafana

### PR DESCRIPTION
**Pull request by aidun**
This PR Fixes #6 

- helm3 only
- default without persistence
- persistence could be enabled with --persistence

Signed-off-by: Markus Hartmann <markush1986@gmail.com>

## Description

Installs Grafana via helm3.

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?

```
kind create cluster
arkade install gafana
# Show all resources
kubectl get all -n grafana

arkade install gafana --persistence
# show the bound pvc
kubectl get pvc -n Grafana
```
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
The PVC should be off by default, thanks for handling that 👍 

Is there ARM support?

**Comment by aidun** [comment]
If you disable sidecar and dashboard support:
```
      sidecar:
        datasources:
          enabled: false
        dasboards:
          enabled: false
```
This images have no arm support.

**Comment by alexellis** [comment]
In which case, don't you think we should prevent the installation on an ARM cluster? See the Istio app as an example.

**Comment by aidun** [comment]
I don't think so, we should disable the features, because we don't use them right now.

**Comment by alexellis** [comment]
Please can you confirm the following is true: this app will not install to ARM? If so, then can you set up platform filtering as per the Istio app?

**Comment by alexellis** [comment]
@Waterdrips Cc

**Comment by Waterdrips** [comment]
This will be very useful! I wonder if we need a `grafana-dashboard` app that can install a grafana dashboard by it's id

**Comment by aidun** [comment]
So if you want this feature , I will exit with errors on non amd64 platforms.  I think we can add the dashboards via flags `—dashboardid`, `—dashboardurl` and `—dashboardfile`.

What about datasources?

**Comment by aidun** [comment]
> Please can you confirm the following is true: this app will not install to ARM? If so, then can you set up platform filtering as per the Istio app?

I can confirm this. The grafana pod do not start because of this sidecar containers. If you disable them, grafana is arm ready.

**Comment by alexellis** [comment]
I don't know how people would want to use this app, I would assume simply installing without sidecars on any platform would be useful, especially for RPi users (from what I am reading, it sounds like that works?)

**Comment by aidun** [comment]
My suggestion: I will implement the original use case. In an additional step, I will do a second PR to provide dashboard and datasource functionality. This will take some time, because I don't know much about Grafana automation. In this time I try to find the maintainer of the sidecar image and ask them to provide arm images. If I find images good, if not this features only work on amd64.

**Comment by aidun** [comment]
I pushed the PR with the original scope.

**Comment by alexellis** [comment]
Hi @aidun due to merging another PR we need a rebase. Please could you look at that? Thanks

**Comment by aidun** [comment]
@alexellis is rebased

File: cmd/apps/grafana_app.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+		fmt.Println(grafanaInstallMsg)
+
+		return nil
+	}
+
+	return grafana
+}
+
+const grafanaInstallMsg = `=======================================================================
+= grafana has been installed.                                   =
```
Comment (alexellis) [inline cmd/apps/grafana_app.go]: I don't think the ASCII lines up properly here?

File: cmd/apps/grafana_app.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+=======================================================================
+
+Get the admin password:
+
+	kubectl get secret --namespace grafana grafana -o jsonpath="{.data.admin-password}" | base64 --decode ; echo
```
Comment (alexellis) [inline cmd/apps/grafana_app.go]: Please only indent `  ` 2x spaces.

File: cmd/apps/grafana_app.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	}
+
+	return grafana
+}
+
+const grafanaInstallMsg = `=======================================================================
+= grafana has been installed.                                   =
+=======================================================================
+
+Get the admin password:
```
Comment (alexellis) [inline cmd/apps/grafana_app.go]: These should be prefixed with `#` as per the other apps.

File: cmd/apps/grafana_app.go (hunk 1 above)
Comment (alexellis) [inline cmd/apps/grafana_app.go]: centre the text please

**Comment by alexellis** [comment]
Thanks for that, approved for merge. You just need to rebase it now - conflict is on cmd/install.go

```
git pull origin master --rebase
```

Note that `--rebase` is really important here, we can't merge merge commits.

**Comment by alexellis** [comment]
@aidun quick question, where is the info message?

**Comment by aidun** [comment]
I missed it. I will add it.

**Pull request merged**